### ISSUE TITLE
False positive for rule 930120 for `test.Enviro`

### ISSUE THREAD

**Issue by aryehb** [coreruleset/coreruleset#3775]
### Description

Data in the form `test.Enviro` will cause rule 930120 to be triggered. This is because the `@pmFromFile` operator is not case sensitive, and the `.env` entry in [lfi-os-files.data](https://github.com/coreruleset/coreruleset/blob/v4.4.0/rules/lfi-os-files.data#L36) does not have word boundries.

### How to reproduce the misbehavior (-> curl call)

```sh
curl --request POST \
  --url https://sandbox.coreruleset.org/ \
  --header 'Content-Type: application/json' \
  --data '{
	"key": "test.Enviro"
}'
```

### Logs

See logs from the sandbox call above.

### Your Environment

* CRS version (e.g., v3.3.4): v4.4.0
* Paranoia level setting (e.g. PL1) : PL1
* ModSecurity version (e.g., 2.9.6): v3.0.12
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): nginx v1.26.1
* Operating System and version: Ubuntu 20 LTS

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by theseion** [comment]
Thank you for the report @aryehb. We'll try to look into this issue soon. We do have lot's of other important open issues to deal with though, so it might take us a while.

Are you aware that you can deal with these kinds of false positives in your own environment by creating an exclusion rule?

**Comment by aryehb** [comment]
Yes, but it's hard to make an exclusion that won't be too broad (e.g. disabling the rule completely) or too narrow (e.g. disabling the rule when it matches `.env` but not `/\b\.env\b/`, which wouldn't help for any other case).

**Comment by Xhoenix** [comment]
You can use a exclusion rule like this:
```
SecRule REQUEST_FILENAME "@endsWith /your-url-endpoint" \
    "id:<fix-me>,\
    phase:1,\
    pass,\
    t:none,\
    nolog,\
    ctl:ruleRemoveTargetById=930120;ARGS:json"
```

This will exclude the entire `lfi-os-files.data` for that specific argument of that specific uri endpoint.

**Comment by EsadCetiner** [comment]
@aryehb Unfortunately a word boundary can't be added without causing false negatives.

I know your concerned about a rule exclusion being too broad, but you can check if a specific target equals ``test.Enviro`` and then disable 930120 for that target.
```
SecRule ARGS:example "@streq test.Enviro" \
    "id:1,\
    phase:2,\
    pass,\
    t:none,\
    ctl:ruleRemoveTargetById=930120;ARGS:example"
```

**Comment by aryehb** [comment]
That data wasn't the only FP I came across. There's an infinite number of possible FPs.

Maybe I could do something like `"!@rx (?<!\w)\.env"`.

**Comment by theseion** [comment]
Rather the other way around:
1. check for `.env` as as substring
2. remove the target (e.g. `ARGS`) for the rule, e.g., `ctl:ruleRemoveTargetById=930120;ARGS:user`

There are two reasons you shouldn't use the snippet you posted above:
1. The `!` is not exactly a negated match. Depending on how the `SecRule` looks, you might get unexpected  results
2. Negative lookahead isn't great for performance

**Comment by aryehb** [comment]
@theseion `.env` is not the only FP for this rule.

We just came across an email with the form `firstname.dockery@host.tld`, where it was being flagged because it matched `.docker`.
